android/src/main/java/ie/gov/tracing/network/DiagnosisKeyDownloader.kt

```diff
@@ -35,8 +35,10 @@ internal class DiagnosisKeyDownloader(private val context: Context) {
         }        
         if (since <= 0) {
             val startIndex = max(0, files.size - fileLimit.toInt())
-            return files.subList(startIndex, fileLimit.toInt()).toTypedArray()
+            val endIndex = min(files.size, startIndex + fileLimit.toInt())
```

Comment:
Should the first arg here be `files.size - 1` or `files.lastIndex` to guarantee the range is within bounds?

Reply:
Also this PR added `import kotlin.math.max` for the `max` call, should it also add `import kotlin.math.min` for these from the last commit?

Reply:
first arg should be the last index we want to hit, in case of an array with 1 element this is 1 rather than 0.

Will add the min import, good catch